src/commands/serverActions.ts

```diff
@@ -56,6 +62,44 @@ export async function serverActions(): Promise<void> {
         await checkConnection(true, undefined, true);
         break;
       }
+      case "switchNamespace": {
+        const allNamespaces: string[] | undefined = await api
+          .serverInfo()
+          .then((data) => data.result.content.namespaces)
+          .catch((reason) => {
+            // Notify user about serverInfo failure
+            vscode.window.showErrorMessage(reason.errorText || `Failed to fetch namespace list`, "Dismiss");
+            return undefined;
+          });
+
+        if (!allNamespaces) {
+          return;
+        }
+        // Handle serverInfo having returned no namespaces
+        if (!allNamespaces.length) {
+          vscode.window.showErrorMessage(`No namespace list returned`, "Dismiss");
+          return;
+        }
+
+        const namespace = await vscode.window.showQuickPick(allNamespaces, {
+          placeHolder: `Namespace on server `,
+          ignoreFocusOut: true,
+        });
+
+        if (!namespace) {
+          return;
+        } else {
```

Comment:
If `namespace` is the same as the currently connected namespace, this code shouldn't be run.

Reply:
Fixed by removing the current namespace from the list of switchable namespaces shown to the user. Doesn't make sense to show the current namespace to the user anyway.